new/chef-upgrade.md

```diff
@@ -0,0 +1,70 @@
+---
+RFC: unassigned
+Author: Thom May <thom@chef.io>
+Status: Draft
+Type: Standards Track
+---
+
+# Chef should be able to upgrade itself
+
+Users should be able to automatically upgrade to the desired version of chef without having to use third party cookbooks. The upgrade should occur without affecting the normal chef client run, ensuring that a consistent state is preserved.
+
+## Motivation
+
+```
+  As an operator, I want chef versions to be consistent across my infrastructure,
+  so that I can ensure my code works correctly
+```
+```
+  As an operator, I want to easily and safely upgrade chef to my desired version,
+  so that I can easily roll out new versions.
+```
+```
+  As a developer, I want our users to get new features and bug fixes in a timely and safe manner,
+  so that everyone's happy.
+```
+
+## Specification
+
+At the end of each chef run, just before the reboot handler, the chef client should check with an update service to discern whether there is a new version of chef. If there is, the client should decide whether or not to upgrade, based on configuration supplied by the administrator.
+Should the decision to upgrade be made, the client will download the new version, and install it appropriately. If a reboot is not triggered, the client run should exit using an appropriate exit code to signal an upgrade.
+
+### Determining the desired chef version
+
+The desired chef version should be provided to the node through a new first class environment or node attribute, named `chef_version`. Once RFC 45 is completed, that attribute will be marked as desired state, and not mutable by the node. The `chef_version` attribute must contain a SemVer parseable version string. A version always takes the form x.y.z, where x, y, and z are decimal numbers that are used to represent major (x), minor (y), and patch (z) versions. One-part (x) and two-part (x.y) versions are allowed.
```

Comment:
I don't think we should specify `node[:chef_version]`. The `chef_upgrade` resource should take a number of properties including version and the user could set that to whatever they want. That could be a fixed string, `node[:chef_version]` or whatever.

You should call out the properties after the next paragraph, e.g. version, channel. When you specify version, you should refer to RFC047#versioning rather than duplicating the explanations here.